common/src/main/java/dev/architectury/core/fluid/ArchitecturyFluidAttributes.java

```diff
@@ -258,16 +293,70 @@ default ResourceLocation getFlowingTexture() {
         return getFlowingTexture(null);
     }
     
+    /**
+     * Returns the overlay texture location of this fluid behind transparent blocks.
+     * <p>
+     * The vanilla water location is {@code "block/water_overlay"}.
+     *
+     * @param state the fluid state, can be {@code null}
+     * @param level the level, can be {@code null}
+     * @param pos   the position, can be {@code null}
+     * @return the texture location, can be {@code null}
+     */
+    @Nullable
+    default ResourceLocation getOverlayTexture(@Nullable FluidState state, @Nullable BlockAndTintGetter level, @Nullable BlockPos pos) {
+        return null;
+    }
+    
+    /**
+     * Returns the overlay texture location of this fluid behind transparent blocks.
+     * <p>
+     * The vanilla water location is {@code "block/water_overlay"}.
+     *
+     * @param stack the fluid stack, can be {@code null}
+     * @return the texture location, can be {@code null}
+     */
+    @Nullable
+    default ResourceLocation getOverlayTexture(@Nullable FluidStack stack) {
+        return null;
+    }
+    
+    /**
+     * Returns the overlay texture location of this fluid behind transparent blocks.
+     * <p>
+     * The vanilla water location is {@code "block/water_overlay"}.
+     *
+     * @return the texture location, can be {@code null}
+     */
+    @Nullable
+    default ResourceLocation getOverlayTexture() {
+        return getOverlayTexture(null);
+    }
+    
     /**
      * Returns the color of the fluid.
      *
      * @param stack the fluid stack, can be {@code null}
      * @param level the level, can be {@code null}
      * @param pos   the position, can be {@code null}
      * @return the color
+     * @deprecated Please use and override {@link #getColor(FluidState, BlockAndTintGetter, BlockPos)}
+     * or {@link #getColor(FluidStack)} instead, this method will be removed in a future version.
      */
     int getColor(@Nullable FluidStack stack, @Nullable BlockAndTintGetter level, @Nullable BlockPos pos);
```

Comment:
This is missing the annotation:
```suggestion
     * @deprecated Please use and override {@link #getColor(FluidState, BlockAndTintGetter, BlockPos)}
     * or {@link #getColor(FluidStack)} instead, this method will be removed in a future version.
     */
    @Deprecated(forRemoval = true)
    int getColor(@Nullable FluidStack stack, @Nullable BlockAndTintGetter level, @Nullable BlockPos pos);
```